app/uk/gov/hmrc/soletraderidentificationfrontend/models/RegistrationStatus.scala

```diff
@@ -23,16 +23,21 @@ sealed trait RegistrationStatus
 
 case class Registered(registeredBusinessPartnerId: String) extends RegistrationStatus
 
-case object RegistrationFailed extends RegistrationStatus
+case class RegistrationFailed(registrationFailures: Option[Array[Failure]]) extends RegistrationStatus
 
 case object RegistrationNotCalled extends RegistrationStatus
 
+case class Failure(code: String, reason: String)
+
 object RegistrationStatus {
   val registrationStatusKey = "registrationStatus"
   val registeredBusinessPartnerIdKey = "registeredBusinessPartnerId"
   val RegisteredKey = "REGISTERED"
   val RegistrationFailedKey = "REGISTRATION_FAILED"
   val RegistrationNotCalledKey = "REGISTRATION_NOT_CALLED"
+  val registrationFailuresKey = "failures"
+
+  implicit val RegistrationFailuresFormat: OFormat[Failure] = Json.format[Failure]
```

Comment:
It is strange because, if I understood well, we use this code to do 2 things ... reading our json from mongodb and reading the incoming json from the BE. (we will never receive from the BE REGISTRATION_NOT_CALLED for example)
That strictly speaking is wrong but I can accept it. What do you think?

Reply:
Yeah correct, we use it to read the json backend returns on the Register call but we also use it for storing/reading from the mongo database. Would be a lot of duplication I imagine if we tried to create two separate models